source/fe/mapping_fe_field.cc

```diff
@@ -58,13 +59,83 @@ MappingFEField<dim,spacedim,VECTOR,DH>::InternalData::InternalData (const Finite
 
 template<int dim, int spacedim, class VECTOR, class DH>
 std::size_t
-MappingFEField<dim,spacedim,VECTOR,DH>::InternalData::memory_consumption () const
+MappingFEField<dim,spacedim,VECTOR,DH>::InternalData::
+memory_consumption () const
 {
   Assert (false, ExcNotImplemented());
   return 0;
 }
 
 
+
+template<int dim, int spacedim, class DH, class VECTOR>
```

Comment:
No. Mapping objects are only ever used through pointers to a base class, never (outside the implementation of the class itself) through the class itself. Consequently, there are no opportunities for inlining and so I thought it is safe to move everything.

In particular, these functions are part of the private interface of the class, so putting them into the .cc file keeps them available for inlining for all possible users (which all reside in the same file anyway).